storage/trace_test.go

```diff
@@ -16,251 +16,142 @@ package storage
+}
 
-				// TODO: Remove setting development env var upon launch.
-				t.Setenv("GO_STORAGE_DEV_OTEL_TRACING", "true")
+func TestStorageTraceEndSpanRecordError(t *testing.T) {
+	ctx := context.Background()
+	te := testutil.NewOpenTelemetryTestExporter()
+	t.Cleanup(func() {
+		te.Unregister(ctx)
+	})
 
-				// Configure the tracer provider and in-memory exporter.
-				e := tracetest.NewInMemoryExporter()
-				tp := sdktrace.NewTracerProvider(sdktrace.WithSyncer(e))
-				defer tp.Shutdown(ctx)
-				otel.SetTracerProvider(tp)
+	// TODO: Remove setting development env var upon launch.
+	t.Setenv("GO_STORAGE_DEV_OTEL_TRACING", "true")
 
-				// Run the library method to test trace instrumentation.
-				err := c.call(ctx, client, &subtest.resources)
-				if err != nil {
-					t.Errorf("%v error: %v", c.name, err)
-				}
+	spanName := "storage.TestTrace.TestRecordError"
+	ctx, _ = startSpan(ctx, spanName)
+	err := &googleapi.Error{Code: http.StatusBadRequest, Message: "INVALID ARGUMENT"}
+	endSpan(ctx, err)
 
-				// Verify trace spans.
-				wantSpan := createWantSpanStub(c.name)
-				checkOTelTraceSpans(t, e, wantSpan)
-				e.Reset()
-			})
-		}
-	}, opts...)
+	spans := te.Spans()
+	gotSpan := spans[0]
+	if len(spans) != 1 {
+		t.Errorf("expected one span, got %d", len(spans))
+	}
+	if got, want := gotSpan.Name, appendPackageName(spanName); got != want {
+		t.Fatalf("got %s, want %s", got, want)
+	}
+	if want := otcodes.Error; gotSpan.Status.Code != want {
+		t.Errorf("got %v, want %v", gotSpan.Status.Code, want)
+	}
 }
 
-func createWantSpanStub(spanName string) tracetest.SpanStub {
+func createWantSpanStub(spanName string, attrs []attribute.KeyValue) tracetest.SpanStub {
 	return tracetest.SpanStub{
-		Name: appendPackageName(spanName),
-		Attributes: []attribute.KeyValue{
-			attribute.String("gcp.client.version", internal.Version),
-			attribute.String("gcp.client.repo", gcpClientRepo),
-			attribute.String("gcp.client.artifact", gcpClientArtifact),
-		},
-		InstrumentationLibrary: instrumentation.Scope{
+		Name:       appendPackageName(spanName),
+		Attributes: attrs,
+		InstrumentationScope: instrumentation.Scope{
 			Name:    "cloud.google.com/go/storage",
 			Version: internal.Version,
 		},
 	}
 }
 
-func checkOTelTraceSpans(t *testing.T, e *tracetest.InMemoryExporter, wantSpan tracetest.SpanStub) {
-	spans := e.GetSpans()
-	if len(spans) == 0 {
-		t.Errorf("Wanted trace spans, got none")
-	}
-	opts := []cmp.Option{
-		cmp.Comparer(spanAttributesComparer),
-	}
-	for _, span := range spans {
-		if diff := testutil.Diff(span, wantSpan, opts...); diff != "" {
-			t.Errorf("diff: -got, +want:\n%s\n", diff)
-		}
-	}
-}
-
 func spanAttributesComparer(a, b tracetest.SpanStub) bool {
 	if a.Name != b.Name {
 		return false
 	}
 	if len(a.Attributes) != len(b.Attributes) {
 		return false
 	}
-	if a.InstrumentationLibrary != b.InstrumentationLibrary {
+	if a.InstrumentationScope != b.InstrumentationScope {
```

Comment:
Why this change?

Reply:
The linter helped pick this up - `InstrumentationLibrary` is deprecated, use `InstrumentationScope` instead
https://github.com/open-telemetry/opentelemetry-go/blob/a9cbc3d8dec7be22c7d3691ca1755f25c1702a1d/sdk/trace/tracetest/span.go#L65-L66